src/api/goods.ts

```
@@ -8,9 +8,18 @@ export function getAll(): Promise<Good[]> {
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll().then(goods => {
+    const sortedGoods = goods.sort((a, b) => {
+      const nameA = a.name.toUpperCase();
+      const nameB = b.name.toUpperCase();
+
+      return nameA.localeCompare(nameB);
+    });
+
+    return sortedGoods.slice(0, 5);
+  });
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll().then(goods => goods.filter(good => good.color === 'red'));
```

This also violates checklist item #3: `getRedGoods` forwards the result of `getAll().then(...)` but does not handle possible fetch errors. If `getAll` fails (network error or non-OK response), `getRedGoods` should either propagate the error in a predictable way or handle it and return a rejected promise / meaningful error object so the UI can display a failure state. Ensure `getAll` implements proper error handling so these helpers will behave correctly.